packages/eslint-plugin/src/rules/no-js-xcss/index.ts

```diff
@@ -0,0 +1,32 @@
+import type { Rule } from 'eslint';
+
+export const noJavaScriptXCSSRule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      recommended: true,
+      description:
+        'The xcss prop is predicated on adhering to the type contract. Using it without TypeScript breaks this contract and thus is not allowed.',
+      url: 'https://github.com/atlassian-labs/compiled/tree/master/packages/eslint-plugin/src/rules/no-js-xcss',
+    },
+    messages: {
+      'no-js-xcss':
+        'Using xcss in JavaScript risks incidents and unintended behaviour when code changes — only use inside TypeScript files',
+    },
+    type: 'problem',
+  },
+  create(context) {
+    const violations = new WeakSet<Rule.Node>();
```

Comment:
where is `violations` used?

Reply:
In the create() function scope to check if a parent had a violation or not, if it does bail out

Reply:
Oops good shout this is from copy paste code. Will remove!

Reply:
Updated